src/gui/plugins/video_recorder/VideoRecorder.qml

```diff
@@ -37,16 +37,67 @@ ToolBar {
         title: "Save the recorded video"
         folder: shortcuts.home
         selectExisting: false
+        property var fileFormat: ""
+        property var selectedFormat: ""
+
+        function getFormat(url) {
+          return (url.slice(url.lastIndexOf(".") + 1))
+        }
+        
         onAccepted: {
-          VideoRecorder.OnSave(fileDialog.fileUrl, fileDialog.selectedNameFilter)
-          close()
+          fileFormat = getFormat(fileUrl.toString())
+          if (fileFormat == fileUrl.toString()) {
+            // no format specified
+            VideoRecorder.OnSave(fileFormat + "." + selectedFormat)
+            close()
+          }
+          else if (fileFormat != selectedFormat){
+            // wrong format specified
+            mismatchDialog.open()
+          } else {
+            // correct format specified
+            VideoRecorder.OnSave(fileUrl)
+            close()
+          }
         }
         onRejected: {
           VideoRecorder.OnCancel()
           close()
         }
     }
 
+    Dialog {
+      id: mismatchDialog 
+      title: "Enconding and filename mismatch"
+      modal: true
+      focus: false
+      width: 700
+      height: 200
+      parent: ApplicationWindow.overlay
+      x: (parent.width - width) / 2
+      y: (parent.height - height) / 2
```

Comment:
@iche033 I'd really appreciate if you could tell me if there's anything wrong with this. I made research and it works for me, but I've got the feeling that on a screen other than my laptop's weirds things could happen. 

Thank you!